modules/gapi/test/common/gapi_video_tests_inl.hpp

```diff
@@ -88,6 +89,34 @@ TEST_P(BuildPyr_CalcOptFlow_PipelineTest, AccuracyTest)
     compareOutputsOptFlow(outOCV, outGAPI);
 }
 
+TEST_P(BackgroundSubtractorMOG2Test, AccuracyTest)
+{
+    initTestDataPath();
+
+    // G-API graph declaration
+    cv::GMat in;
+    cv::GMat out = cv::gapi::video::GBackgroundSubtractorMOG2::on(in, -1);
+    // Preserving 'in' in output to have possibility to compare with OpenCV reference
+    cv::GComputation c(cv::GIn(in), cv::GOut(cv::gapi::copy(in), out));
+
+    // G-API compilation of graph for streaming mode
+    const auto pkg = cv::gapi::kernels<GCPUBackgroundSubtractorMOG2>();
+    auto gapiBackSub = c.compileStreaming(cv::compile_args(pkg, cv::gapi::video::BackgroundSubtractorParams{500, 16, true}));
+
+    //auto gapiBackSub = c.compileStreaming(getCompileArgs());
+    // Testing G-API Background Substractor in streaming mode
+    gapiBackSub.setSource(gapi::wip::make_src<cv::gapi::wip::GCaptureSource>
+        (findDataFile(filePath1)));
+    // Allowing 1% difference of all pixels between G-API and reference OpenCV results
+    testBackSubInStreaming(gapiBackSub, 1);
+
+    // Additionally, test the case when the new stream happens
+    gapiBackSub.setSource(gapi::wip::make_src<cv::gapi::wip::GCaptureSource>
+        (findDataFile(filePath2)));
+    // Allowing 5% difference of all pixels between G-API and reference OpenCV results
+    testBackSubInStreaming(gapiBackSub, 5);
```

Comment:
Is difference worse because of greater resolution of the file? Just interesting :)

Reply:
Difference between GAPI and OCV results should be smaller than 5%. Here nothing is said about the file resolution.

Reply:
Rework test. Please take a look.

Reply:
Seems OK